examples/kitchen-sink.yaml

```diff
@@ -91,6 +91,10 @@ logger_provider:
             #
             # Environment variable: OTEL_EXPORTER_OTLP_INSECURE, OTEL_EXPORTER_OTLP_LOGS_INSECURE
             insecure: false
+            # Configure retry policy.
+            retry:
+              # Configure retry disabled.
+              disabled: false
```

Comment:
What do you think of @carlosalberto's [suggestion](https://github.com/open-telemetry/opentelemetry-specification/issues/4138#issuecomment-2239590144) to have a `strategy` field, defaulting to `exponential_retry` and supporting `none`.

Reply:
I really like it, and I think it could resolve my issues. If we allow for future retry strategies, including custom/3rd party strategies, then SIGs have the freedom to provide their own configuration to suit their specific implementation.
If the official exponential retry strategy is locked down in spec, SIGs can migrate across to it at their leisure (w.r.t using it with file-based configuration).

How about treating retry policy similarly to how we treat a `Sampler` - it's configuration is optional and defaults to `exponential_retry`, at most one can be provided, and custom is allowed? Something like:

```yaml
retry:
  disabled: false
  <policy-name>:
    property: value
```

`disabled` could actually be replaced by a ` noop`/`none`/`noretry` policy, which might make the interface between an exporter and the retry policy a little cleaner.